tlittenberg/lisacattools
# Refactoring

Refactoring : 

- use virtualenv
- source code (plugin architecture to manage, code refactoring) 
- project structure:
     * COTS are identified in requirements.txt
     * COTS development are identified in requirements-dev.txt
     * COTS for demo are identified in requirements-demo.txt
     * Test directory contains tests
     * scripts directory contains quality tools that are performed when  commit is run
     * docs contains everything to generate the documentation
     * tutorial contains the notebook and the data
- add quality tests
     * reformatting the code, Cyclomatic Complexity, Maintenability, Tests.  If the verification of one of these tools is not correct, the commitment is refused. To avoid checks: git commit --no-verify
- Use Makefile to run tasks (install cots, build web site, ...)
